jsmiddleware/src/main/java/com/segment/jsmiddleware/EdgeFunctionMiddlewareProcessor.kt

```diff
@@ -2,80 +2,48 @@ package com.segment.jsmiddleware
-    private val client = OkHttpClient()
+class EdgeFunctionMiddlewareProcessor(private val listener: EdgeFunctionMiddlewareProcessorListener) {
 
-    ////
-    // Public Methods
     @Throws(Exception::class)
-    fun configureLocalFile(mobContext: Context, localFile: String) {
+    fun configureCachedFile(mobContext: Context) {
+        listener.jsRetrieved(getCachedFileContents(mobContext))
+    }
+
+    @Throws(Exception::class)
+    fun configureFallbackFile(mobContext: Context, localFile: String) {
         val localJSMiddlewareInputStream = mobContext.assets.open(localFile)
-        val result = IOUtils.toString(localJSMiddlewareInputStream, StandardCharsets.UTF_8)
+        val result = readInputStream(localJSMiddlewareInputStream)
         listener.jsRetrieved(result)
     }
 
     @Throws(Exception::class)
-    fun downloadFileAsync(mobContext: Context, downloadUrl: String?) {
-
-        // Grab settings first
-//        try {
-//            val settings = fetchSettings()
-//        } catch (e: Exception) {
-//            Log.e("JSMiddleware", "Could not parse settings")
-//        }
-        val request = Request.Builder().url(downloadUrl!!).build()
-        client.newCall(request).enqueue(object : Callback {
-            override fun onFailure(call: Call, e: IOException) {
-                e.printStackTrace()
-            }
-
-            @Throws(IOException::class)
-            override fun onResponse(call: Call, response: Response) {
-                if (!response.isSuccessful) {
-                    throw IOException("Failed to download file: $response")
-                }
-                try {
-                    response.body.use { responseBody ->
+    fun downloadFile(mobContext: Context, downloadUrl: String) {
+        var connection: ConnectionUtils.Connection? = null
+        try {
+            connection = downloadEdgeFunctionBundle(downloadUrl)
+            if (connection.inputStream != null) {
+                val inputStream = connection.inputStream!!
+                val function = readInputStream(inputStream)
 
-                        // Write the file out
-                        if (responseBody != null) {
-                            val fos = FileOutputStream(jsDownloadFullPath(mobContext, "blah.js"))
-                            fos.write(responseBody.bytes())
-                            fos.close()
-                        }
+                Log.d("JSMiddleware", "New Edge Function downloaded, will be used on next app restart")
 
-                        // Let the executor know the file has been saved
-                        listener.jsRetrieved(getLocalFile(mobContext))
-                    }
-                } catch (e: Exception) {
-                    Log.e("JSMiddleware", "Could not parse file body")
-                }
+                val fos = FileOutputStream(jsDownloadFullPath(mobContext, "blah.js"))
```

Comment:
Next step we may want to remove hard coded blah.js

Reply:
yea i wasnt sure about your thought-process here, so i didnt wanna change anything